numpy/lib/tests/test_nanfunctions.py

```diff
@@ -83,21 +83,23 @@ def test_result_values(self):
             res = nf(_ndat, axis=1)
             assert_almost_equal(res, tgt)
 
-    def test_allnans(self):
-        mat = np.array([np.nan]*9).reshape(3, 3)
-        for f in self.nanfuncs:
-            for axis in [None, 0, 1]:
-                with warnings.catch_warnings(record=True) as w:
-                    warnings.simplefilter('always')
-                    assert_(np.isnan(f(mat, axis=axis)).all())
-                    assert_(len(w) == 1, 'no warning raised')
-                    assert_(issubclass(w[0].category, RuntimeWarning))
-            # Check scalars
-            with warnings.catch_warnings(record=True) as w:
-                warnings.simplefilter('always')
-                assert_(np.isnan(f(np.nan)))
-                assert_(len(w) == 1, 'no warning raised')
-                assert_(issubclass(w[0].category, RuntimeWarning))
+    @pytest.mark.parametrize("axis", [None, 0, 1])
+    @pytest.mark.parametrize("dtype", np.typecodes["AllFloat"])
+    @pytest.mark.parametrize("array", [
+        np.full((3, 3), np.nan),
+        np.array(np.nan),
+    ], ids=["0d", "2d"])
+    def test_allnans(self, axis, dtype, array):
+        if axis is not None and array.ndim == 0:
+            pytest.skip(f"`axis != None` not supported for 0d arrays")
+
+        array = array.astype(dtype)
+        match = "All-NaN slice encountered"
+        for func in self.nanfuncs:
+            with pytest.warns(RuntimeWarning, match=match):
+                out = f(array, axis=axis)
```

Comment:
Is there a reason why `nanmin` warns for all-nan slices but `nanargmin` raises?

Both operations lack an identity, and their `argmin`/`min` counterparts both raise when encountering size-zero arrays. Based on this I'd expect `nanmin` to do the same for all-nan arrays.

Reply:
The thinking for `nanmin` was that returning nan in that case was a way to mark that result as invalid, perhaps unexpectedly, while the other minimums in the reduction might be correct.  That differs from the behavior for an empty array which reliably fails on _all_ the minimum values. Effectively, removing the nans makes for a ragged array  with only some of the slices possibly empty. There is not easy way to do the same for `argmin`. I suppose the index could point to a nan value, but it wouldn't be obvious which indexes were bad.